lib/controllers/root-controller.js

```diff
@@ -362,13 +367,15 @@ export default class RootController extends React.Component {
   }
 
   @autobind
-  async initializeRepo() {
+  async initializeRepo(initDialogPath) {
     if (this.props.activeWorkingDirectory) {
       await this.acceptInit(this.props.activeWorkingDirectory);
-      return;
+      return null;
     }
 
-    this.setState({initDialogActive: true});
+    return new Promise(resolve => {
+      this.setState({initDialogActive: true, initDialogPath, initDialogResolve: resolve});
+    });
```

Comment:
> Is it possible for this setState() call to be triggered twice before the first resolution callback is triggered... ?

This is a good question and a nice catch. As things are now, I think it's highly unlikely that we run into this issue. We only get into this method from the user clicking the "create repo" button, which opens the dialog, and closing the dialog resolves the promise. There's no opportunity for the user to click the "create repo" button again before the dialog is closed.

The only way I can see this method being called before the resolution is if we add a command for initializing a repo (which seems reasonable). So to address this I'll add a guard at the top of the method and if `this.state.initDialogActive` is already true, we'll do an early return from the method.